src/sentry/deletions/defaults/group.py

```diff
@@ -76,16 +104,9 @@ def chunk(self) -> bool:
                 ]
             )
 
-        group_ids = []
-        project_groups = defaultdict(list)
-        for group in self.groups:
-            project_groups[group.project_id].append(group.id)
-            group_ids.append(group.id)
-        project_ids = list(project_groups.keys())
```

Comment:
This block is moved to `set_group_and_project_ids`, which is called within the class initialization. There's no need to keep calculating it on every chunk call.
<img width="522" alt="image" src="https://github.com/user-attachments/assets/113caccf-616c-4824-b5cf-d2a19be4e4a5">